cmake/platform/linux/wayland.cmake

```diff
@@ -0,0 +1,17 @@
+set(PLATFORM_REQUIRED_DEPS EGL Waylandpp LibDRM Xkbcommon)
+set(PLATFORM_OPTIONAL_DEPS VAAPI)
+
+set(WAYLAND_RENDER_SYSTEM "" CACHE STRING "Render system to use with Wayland: \"gl\" or \"gles\"")
+
+if(WAYLAND_RENDER_SYSTEM STREQUAL "gl")
+  list(APPEND PLATFORM_REQUIRED_DEPS OpenGl)
+elseif(WAYLAND_RENDER_SYSTEM STREQUAL "gles")
+  list(APPEND PLATFORM_REQUIRED_DEPS OpenGLES)
+else()
+  message(SEND_ERROR "You need to decide whether you want to use GL- or GLES-based rendering in combination with the Wayland windowing system. Please set WAYLAND_RENDER_SYSTEM to either \"gl\" or \"gles\". For normal desktop systems, you will usually want to use \"gl\".")
```

Comment:
In case it wasn't clear. We already have options for selecting gl or gles, those should be used instead of inventing another option that does the same thing.